### PR TITLE
smug: add man page

### PR DESCRIPTION
**Pull request by juboba**
###### Motivation for this change
Man page for `smug` is missing.

###### Things done
Added man page to the derivation so it is available to users.

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ckiee** [comment]
Result of `nixpkgs-review pr 162801` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>smug</li>
  </ul>
</details>

**Comment by c0bw3b** [comment]
It's clearer to fetch upstream tag rather than a commit id to see immediately it is indeed the `version` we claim it is.
You're also missing ldflags to pass to Go build so the tool report its version instead of "dev".
As for installing  the man page, we have a [convenient shell function `installManPage`](https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/stdenv.chapter.md#installshellfiles-installshellfiles) making this easier.

---

You would end up with:

```nix
{ lib, buildGoModule, fetchFromGitHub, installShellFiles }:

buildGoModule rec {
  pname = "smug";
  version = "0.2.7";

  subPackages = [ "." ];

  src = fetchFromGitHub {
    owner = "ivaaaan";
    repo = "smug";
    rev = "v${version}";
    sha256 = "178125835dhnaq9k42yv4pfxpyhgb5179wrxkimb59fy0nk8jzx8";
  };

  vendorSha256 = "1rba5rpvlr8dyhj145b5i57pm4skfpj3vm7vydkn79k6ak6x985x";

  nativeBuildInputs = [ installShellFiles ];

  ldflags = [ "-X=main.version=${version}" ];

  postInstall = ''
    installManPage ./man/man1/smug.1
  '';

  meta = with lib; {
    homepage = "https://github.com/ivaaaan/smug";
    description = "Smug - tmux session manager";
    license = licenses.mit;
    maintainers = with maintainers; [ juboba ];
  };
}
```

Feel free to reuse that and make it your own. :)

**Comment by juboba** [comment]
@c0bw3b Thanks a lot!

File: pkgs/tools/misc/smug/default.nix [hunk 1]
```diff
@@ -9,12 +9,20 @@ buildGoModule rec {
...
-    rev = "3399f02a6e01324f5bb881f6b049c9e8d94733ee";
+    rev = "v${version}";
     sha256 = "178125835dhnaq9k42yv4pfxpyhgb5179wrxkimb59fy0nk8jzx8";
   };
 
   vendorSha256 = "1rba5rpvlr8dyhj145b5i57pm4skfpj3vm7vydkn79k6ak6x985x";
 
+  nativeBuildInputs = [ installShellFiles ];
+
+  ldflags = [ "-X=main.version=${version}" ];
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/smug/default.nix:20]:
```suggestion
  ldflags = [ "-s" "-w" "-X=main.version=${version}" ];
```

**Pull request merged**